File: code_combat/4_Backwoods_Forest/411-Forest_Flower_Grove/forest_flower_grove.py
from math import pi, sin, cos

def circle(angle, condition, step):
    angle = pi * angle
    while angle < pi * condition:
        hero.moveXY(hero.pos.x + cos(angle),hero.pos.y + sin(angle))
        angle += pi / step


circle(0.5, 1.5, 96)
circle(1.5, 2.5, 124)
circle(0.5, 1.5, 96)
circle(1.5, 20, 68)

File: projects/code_combat/4_Backwoods_Forest/411-Forest_Flower_Grove/forestFlowerGrove.js
// This level is a place for making flower art.
// The real goal is to experiment and have fun!
// If you draw something with at least 1000 flowers, you will *succeed* at the level

var angle = Math.PI / 2;
while (angle < Math.PI * 1.5) {
    hero.moveXY(hero.pos.x + Math.cos(angle), hero.pos.y + Math.sin(angle));
    angle += Math.PI / 96;
}


angle = Math.PI * 1.5;
while (angle < Math.PI * 2.5) {
    hero.moveXY(hero.pos.x + Math.cos(angle), hero.pos.y + Math.sin(angle));
    angle += Math.PI / 124;
}

angle = Math.PI / 2;
while (angle < Math.PI * 1.5) {
    hero.moveXY(hero.pos.x + Math.cos(angle), hero.pos.y + Math.sin(angle));
    angle += Math.PI / 96;
}

angle = Math.PI * 1.5;
while (true) {
    hero.moveXY(hero.pos.x + Math.cos(angle), hero.pos.y + Math.sin(angle));
    angle += Math.PI / 68;
}
